packages/@ember/-internals/glimmer/tests/integration/event-dispatcher-test.js

```diff
@@ -616,6 +617,179 @@ if (jQueryDisabled) {
       }
     }
   );
+
+  moduleFor(
+    'EventDispatcher - transitionTo from <a> tags',
+    class extends ApplicationTestCase {
+      constructor() {
+        super();
+
+        this.router.map(function () {
+          this.route('about');
+          this.route('blog', function () {
+            this.route('post', { path: ':id' }, function () {
+              this.route('comments', { path: ':dynamic-param' });
+            });
+          });
+        });
+
+        this.addTemplate(
+          'application',
+          `
+          <nav>
+            <a class="home" href="/">Home</a>
+            <a class="about" href="/about">About</a>
+            <a class="blog" href="/blog">Blog</a>
+            <a class="post" href="/blog/1">Post</a>
+            <a class="deep-nest" href="/blog/1/some-param">Deeply Nested</a>
+            <a class="QPs" href="/blog/1/some-param?applicationQP=1&q=2&foo=3">with QPs</a>
+            <a class="QPHash" href="/blog/1/some-param?applicationQP=1&q=2&foo=3#target">with QPs and hash</a>
+            <a class="hash" href="/blog/1/some-param#target">with hash</a>
```

Comment:
should we add `bad` cases?

`<a href="javascript:void(0)" id="loginlink">login</a>`
`<a heef="#id-on-page">to some block</a>`
`<a href="#" onclick="return false;">hello</a>`